atlasdb-timelock-server/src/main/java/com/palantir/atlasdb/timelock/TimeLockServer.java

```diff
@@ -150,23 +150,24 @@ private static TimeLockServices createInvalidatingTimeLockServices(
                 TimeLockServices.class);
     }
 
-    private static Transport createTransport(Optional<SslConfiguration> optionalSecurity) {
+    private static Transport createTransport(Optional<AtomixSslConfiguration> optionalSecurity) {
         NettyTransport.Builder transport = NettyTransport.builder();
 
         if (!optionalSecurity.isPresent()) {
             return transport.build();
         }
 
-        SslConfiguration security = optionalSecurity.get();
+        AtomixSslConfiguration security = optionalSecurity.get();
         transport.withSsl()
-                .withTrustStorePath(security.trustStorePath().toString());
+                .withTrustStorePath(security.sslConfiguration().trustStorePath().toString())
+                .withTrustStorePassword(security.trustStorePassword());
 
-        if (security.keyStorePath().isPresent()) {
-            transport.withKeyStorePath(security.keyStorePath().get().toString());
+        if (security.sslConfiguration().keyStorePath().isPresent()) {
+            transport.withKeyStorePath(security.sslConfiguration().keyStorePath().get().toString());
         }
 
-        if (security.keyStorePassword().isPresent()) {
-            transport.withKeyStorePassword(security.keyStorePassword().get());
+        if (security.sslConfiguration().keyStorePassword().isPresent()) {
+            transport.withKeyStorePassword(security.sslConfiguration().keyStorePassword().get());
         }
```

Comment:
`SslConfiguration` has an invariant that either both or neither of `keyStorePath` and `keyStorePassword` are present. I'm thinking we could have a helper `isKeystoreSpecified()` or something that just checks if the path is present, and then adds both fields.

Or maybe I'm trying to be too clever here - feel free to push back.